Sources/Honeycomb/UIKit/UIViewController.swift

```diff
@@ -4,6 +4,31 @@
     import UIKit
 
     extension UIViewController {
+        var storyboardId: String? {
+            return value(forKey: "storyboardIdentifier") as? String
+        }
+
+        private var viewName: String {
+            return self.view.accessibilityIdentifier ?? self.storyboardId ?? self.title
```

Comment:
I think this should be:

```swift
return self.view.accessibilityIdentifier
    ?? self.title
    ?? self.storyboardId
    ?? NSStringFromClass(type(of: self))
```

From the debugger:
```
(lldb) po self.storyboardId
▿ Optional<String>
  - some : "UIViewController-Y6W-OH-hqX"
(lldb) po self.title
▿ Optional<String>
  - some : "UIKit Screen"
```

So I do think the `title` is better than the questionable `storyboardId`.

Reply:
The problem with this is that `UINavigationController` acquires the title of its root stack, so you end up with this fun duplication of names:

![image](https://github.com/user-attachments/assets/8bcaf185-5608-4d51-9a47-f263c8239e8d)
![image](https://github.com/user-attachments/assets/79e71ad9-a758-4dcb-8142-546daebe3a13)
(the first `UIKit Menu` refers to the root `UINavigationController` controlling the whole thing, the second refers to the view it's displaying)

whereas doing `storyboardId` first gives us

![image](https://github.com/user-attachments/assets/bb1d5250-2d41-4b14-8238-8c49f3e40ab6)
![image](https://github.com/user-attachments/assets/04c495ff-2d99-4dac-93f0-2eaaf0b4bf34)

Reply:
I don't know how confusing this would be for our users, or maybe I configured the storyboard incorrectly...

Reply:
I spent some time thinking about this, but didn't reach any conclusion. It seems that there are several conflicting constraints:
1. Listing the `UINavigationController` in the path with the same name as the child view kinda sucks.
2. Listing `"UIViewController-Y6W-OH-hqX"` in the path instead of the actual title kinda sucks.
3. Relying too much on the undocumented, unsupported `storyboardId` field is relatively risky.

So then I think we have these options:
1. Prefer `storyboardId` to `title`, and encourage use of the `accessibilityIdentifier` override.
2. Prefer `title` to `storyboardId`, and accept that part of the path is not useful. If we switch to using just the last thing in the path as the `screen.name`, maybe this is less of an issue?
3. Special-case our logic so that for `UINavigationController`, we prefer `storyboardId`, while for other controllers, we prefer `title`. We'd have to investigate other common controllers to see how they behave, although we should probably do that anyway.

I think I prefer Option 3 > 2 > 1, but I'm open to discussion.